src/slurm_plugin/slurm_resources.py

```diff
@@ -318,8 +343,11 @@ def is_healthy(self, terminate_drain_nodes, terminate_down_nodes, log_warn_if_un
 
     def is_state_healthy(self, terminate_drain_nodes, terminate_down_nodes, log_warn_if_unhealthy=True):
         """Check if a slurm node's scheduler state is considered healthy."""
-        # Check to see if node is in DRAINED, ignoring any node currently being replaced
-        if self.is_drained() and terminate_drain_nodes:
+        # Check if node is rebooting: if so, the node is healthy
+        if self.is_node_rebooting():
```

Comment:
Can a node in `is_reboot_issued` unhealthy?

Reply:
> Can a node in `is_reboot_requested` unhealthy?

If by "unhealthy" we mean either "drained" or "down" (which are the two conditions we consider unhealthy for compute nodes), then yes, and that's exactly why we need that exclusion.

> If I understand correctly, reboot nodes are always in down or drained state, right?
These are the main cases:
1. `MIXED/ALLOCATED/IDLE+REBOOT_REQUESTED`: this is the result of `scontrol reboot <node>`. If a job is running or the node is selected for a job scheduled for the future, the scheduler won't trigger the reboot yet (it's up to the scheduler to decide when it's a good time to reboot the node).
2. `MIXED/ALLOCATED+DRAIN+REBOOT_REQUESTED`: this is the result of `scontrol reboot asap <node>`. The `asap` flag means that the node is set to `DRAIN`, so that it doesn't accept any more jobs after the one currently running (meaning that the node will be rebooted as soon as the running job is over).
3. `DOWN+REBOOT_ISSUED`: this is the state after the reboot has been issued for case 1.
4. `DOWN+DRAIN+REBOOT_ISSUED`: this is the state after the reboot has been issued for case 2.

The cases we need to consider in `is_state_healthy()` are 2, 3 and 4. Case 1 is not problematic in our current logic.

> Shall we add some unit tests cases to check is_node_rebooting?

We could, but we already have unit tests for `is_state_healthy()`, which always calls `is_node_rebooting()`. But it doesn't cost much, so we can add them.

Reply:
I added some new unit tests for the `is_rebooting()` method (also, I changed name from `is_node_rebooting()` to `is_rebooting()`.